fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/statemachine/ReplicaStateMachine.java

```diff
@@ -206,19 +217,12 @@ private void doHandleStateChanges(
                             if (currentState != ReplicaState.NewReplica) {
                                 TableBucket tableBucket = replica.getTableBucket();
                                 String partitionName;
-                                if (tableBucket.getPartitionId() != null) {
-                                    partitionName =
-                                            coordinatorContext.getPartitionName(
-                                                    tableBucket.getPartitionId());
-                                    if (partitionName == null) {
-                                        LOG.error(
-                                                "Can't find partition name for partition: {}.",
-                                                tableBucket.getPartitionId());
-                                        logFailedSateChange(replica, currentState, targetState);
-                                        return;
-                                    }
-                                } else {
-                                    partitionName = null;
+                                try {
+                                    partitionName = getPartitionName(tableBucket);
+                                } catch (PartitionNotExistException e) {
+                                    LOG.error(e.getMessage());
+                                    logFailedSateChange(replica, currentState, targetState);
+                                    return;
```

Comment:
should be `continue`? Otherwise, some LeaderAndIsr changes may not be notified.

Reply:
Do you mean the LeaderAndIsr change for other `validReplicas`? If so, `return` will just skip the current replica, and the other valida replicas will still go through the loop.  
Also, user `continue` wiil make my IDE error the message `Continue outside of loop`

Reply:
Sorry, I mean this `return` https://github.com/alibaba/fluss/pull/664/files/33facf0e8b96a793ecb717a07affad2c855dcf4d#diff-83fa38b29066726ca71790bb02d2e2734d8ebfbe24c23806eb55a53d64f6c824R289